sdk/core/azure-core/CHANGELOG.md

```diff
@@ -6,6 +6,10 @@
 
 - Added support for HTTP conditional requests `MatchConditions` and `RequestConditions`.
 
+### Breaking Changes
+
+- Remove `Context::CancelWhen()`.
```

Comment:
Why did we decide to remove this? Is it only because we don't have any callers within the SDK, or do we expect end-customers won't benefit from knowing when a particular context will get cancelled, either?

nit: Can we do this separately from a test improvement PR?

Reply:
I want to close the issue and you tied that to this method :).  There isn't a scenario for using this method so now is a good time to remove it.  If a customer wants a specific timeout for a call they will create a new context with that specific deadline.

Reply:
>  If a customer wants a specific timeout for a call they will create a new context with that specific deadline.

But what happens if some other thread who owns one of the parent contexts, calls cancel and hence changes the `time_point`? How is the owner of the child context know what the cancel time is without this method?

If I. as the caller, just created my context using deadline, I have this value (for my specific context only). But if I create it using value, or rely on the parent cancellation to cancel it, it would be useful for me to know when my context will get cancelled. Without calling `CancelWhen()`, how will I know the actual value?

Reply:
I'm OK with removing this (for now). The purpose is that it allows a caller to find out when the nearest cancellation time is and then this can be used when calling another function. For example, if the cancellation is in 5 minutes, then a thread sleep that might take more than this can short-circuit this (we were going to use this in our retry logic).

Another example is that Azure Storage REST APIs support a timeout query parameter. When making a request, we could set this query parameter based on the time remaining before cancellation - we don't have code that does this today.